Framework/Algorithms/src/ConvertUnitsUsingDetectorTable.cpp

```diff
@@ -276,29 +120,52 @@ void ConvertUnitsUsingDetectorTable::convertViaTOF(
         "DetectorParameter TableWorkspace is not defined correctly.");
   }
 
-  // Now let's read them into some vectors.
-  auto l1Column = paramWS->getColVector<double>("l1");
-  auto l2Column = paramWS->getColVector<double>("l2");
-  auto twoThetaColumn = paramWS->getColVector<double>("twotheta");
-  auto efixedColumn = paramWS->getColVector<double>("efixed");
-  auto emodeColumn = paramWS->getColVector<int>("emode");
-  auto spectraColumn = paramWS->getColVector<int>("spectra");
-
-  EventWorkspace_sptr eventWS =
-      boost::dynamic_pointer_cast<EventWorkspace>(outputWS);
-  assert(static_cast<bool>(eventWS) == m_inputEvents); // Sanity check
+  // Now let's take a reference to the vectors.
+  const auto &l1Column = paramWS->getColVector<double>("l1");
+  const auto &l2Column = paramWS->getColVector<double>("l2");
+  const auto &twoThetaColumn = paramWS->getColVector<double>("twotheta");
+  const auto &efixedColumn = paramWS->getColVector<double>("efixed");
+  const auto &emodeColumn = paramWS->getColVector<int>("emode");
+  auto &spectraColumn = paramWS->getColVector<int>("spectra");
 
   Progress prog(this, 0.2, 1.0, m_numberOfSpectra);
   int64_t numberOfSpectra_i =
       static_cast<int64_t>(m_numberOfSpectra); // cast to make openmp happy
 
   // Get the unit object for each workspace
-  Kernel::Unit_const_sptr outputUnit = outputWS->getAxis(0)->unit();
-
+  Kernel::Unit_const_sptr outputUnit = m_outputUnit;
   std::vector<double> emptyVec;
   int failedDetectorCount = 0;
 
-  // ConstColumnVector<int> spectraNumber = paramWS->getVector("spectra");
+  // Perform Sanity Validation before creating workspace
```

Comment:
@NickDraper Just for my understanding: I could not find an equivalent for the code block performing these checks in the old version, i.e., these are new checks you added?